### DISCUSSION TITLE
libMesh Update (2021.11.11)

### DISCUSSION THREAD

**Discussion by cticenhour** [idaholab/moose#19532] [News]
A new libMesh made its way to the MOOSE master branch in November. For those using the conda packages, please perform
```
conda update --all
```
in your MOOSE conda environment. The appropriate `moose-libmesh` package version should be `2021.11.11`. For those using the the libMesh submodule, please run in the moose/scripts directory
```
./update_and_rebuild_libmesh.sh
```
after updating your copy of MOOSE. 

More details on the content of this update can be found in [the November 2021 Newsletter](https://mooseframework.inl.gov/newsletter/2021_11.html).

**Comment by GiudGiud** [comment]
**UPDATE TO THE UPDATE BELOW**

When `conda (or mamba) update --all` fails to correctly update libmesh, you can force an update to the right package with:

```
mamba update moose-libmesh=<latest-version-date>
```

The current version date, in early February 2022, is `2022.01.19`. You will find the latest version date on the Discussions announcement.
If this fails, please follow the instructions below.

**UPDATE**

`conda update --all` has failed to correctly update libmesh on systems with higher versions of python that we currently allow. This will cause MOOSE build failures. 

To confirm this issue is affecting you, please type `conda list | grep libmesh` and look at the version. If different from 2021.11.11 (or the current latest if you are looking at this comment in 2022), this is affecting your system. 

The solution is to remove the moose conda environment and re-install it. 
```
conda deactivate
conda env remove -n moose
conda install mamba -y
conda create --name moose -q -y
conda activate moose
mamba install moose-libmesh moose-tools
```

**Reply by mangerij** [reply]
Also, your system might need a reboot after these commands :)
